@samccann Thanks for review.
Unfortunately, it's still necessary to ignore this module since hypervisor choices in argument_spec also include lowercase hypervisor names.

~~~
ERROR: lib/ansible/modules/cloud/cloudstack/cs_cluster.py:0:0: E326 argument_spec for 'hypervisor' defines choices as '['KVM', 'kvm', 'VMware', 'vmware', 'BareMetal', 'baremetal', 'XenServer', 'xenserver', 'LXC', 'lxc', 'HyperV', 'hyperv', 'UCS', 'ucs', 'OVM', 'ovm', 'Simulator', 'simulator']' but documentation defines choices as '['KVM', 'VMware', 'BareMetal', 'XenServer', 'LXC', 'HyperV', 'UCS', 'OVM']'
~~~

--- a/lib/ansible/modules/cloud/cloudstack/cs_account.py
+++ b/lib/ansible/modules/cloud/cloudstack/cs_account.py
@@ -101,9 +101,8 @@ extends_documentation_fragment: cloudstack
 '''
 
 EXAMPLES = '''
-# create an account in domain 'CUSTOMERS'
-- local_action:
-    module: cs_account
+- name: create an account in domain 'CUSTOMERS'
+  cs_account:
     name: customer_xy
     username: customer_xy
     password: S3Cur3
@@ -112,50 +111,51 @@ EXAMPLES = '''
     email: john.doe@example.com
     domain: CUSTOMERS
     role: Domain Admin
+  delegate_to: localhost
 
-# Lock an existing account in domain 'CUSTOMERS'
-- local_action:
-    module: cs_account
+- name: Lock an existing account in domain 'CUSTOMERS'
+  cs_account:
     name: customer_xy
     domain: CUSTOMERS
     state: locked
+  delegate_to: localhost
 
-# Disable an existing account in domain 'CUSTOMERS'
-- local_action:
-    module: cs_account
+- name: Disable an existing account in domain 'CUSTOMERS'
+  cs_account:
     name: customer_xy
     domain: CUSTOMERS
     state: disabled
+  delegate_to: localhost
 
-# Enable an existing account in domain 'CUSTOMERS'
-- local_action:
-    module: cs_account
+- name: Enable an existing account in domain 'CUSTOMERS'
+  cs_account:
     name: customer_xy
     domain: CUSTOMERS
     state: enabled
+  delegate_to: localhost
 
-# Remove an account in domain 'CUSTOMERS'
-- local_action:
-    module: cs_account
+- name: Remove an account in domain 'CUSTOMERS'
+  cs_account:
     name: customer_xy
     domain: CUSTOMERS
     state: absent
+  delegate_to: localhost
 
-# Create a single user LDAP account in domain 'CUSTOMERS'
-- local_action:
-    module: cs_account
+- name: Create a single user LDAP account in domain 'CUSTOMERS'
+  cs_account:
     name: customer_xy
     username: customer_xy
     domain: CUSTOMERS
     ldap_domain: cn=customer_xy,cn=team_xy,ou=People,dc=domain,dc=local
+  delegate_to: localhost
 
-# Create a LDAP account in domain 'CUSTOMERS' and bind it to a LDAP group
-- local_action:
-    module: cs_account
+- name: Create a LDAP account in domain 'CUSTOMERS' and bind it to a LDAP group
+  cs_account:
     name: team_xy
     username: customer_xy
     domain: CUSTOMERS
     ldap_domain: cn=team_xy,ou=People,dc=domain,dc=local
+  delegate_to: localhost
 '''
 
 RETURN = '''

--- a/lib/ansible/modules/cloud/cloudstack/cs_affinitygroup.py
+++ b/lib/ansible/modules/cloud/cloudstack/cs_affinitygroup.py
@@ -59,17 +59,17 @@ extends_documentation_fragment: cloudstack
 '''
 
 EXAMPLES = '''
-# Create a affinity group
-- local_action:
-    module: cs_affinitygroup
+- name: Create a affinity group
+  cs_affinitygroup:
     name: haproxy
     affinity_type: host anti-affinity
+  delegate_to: localhost
 
-# Remove a affinity group
-- local_action:
-    module: cs_affinitygroup
+- name: Remove a affinity group
+  cs_affinitygroup:
     name: haproxy
     state: absent
+  delegate_to: localhost
 '''
 
 RETURN = '''

--- a/lib/ansible/modules/cloud/cloudstack/cs_cluster.py
+++ b/lib/ansible/modules/cloud/cloudstack/cs_cluster.py
@@ -112,34 +112,34 @@ extends_documentation_fragment: cloudstack
 '''
 
 EXAMPLES = '''
-# Ensure a cluster is present
-- local_action:
-    module: cs_cluster
+- name: Ensure a cluster is present
+  cs_cluster:
     name: kvm-cluster-01
     zone: ch-zrh-ix-01
     hypervisor: KVM
     cluster_type: CloudManaged
+  delegate_to: localhost
 
-# Ensure a cluster is disabled
-- local_action:
-    module: cs_cluster
+- name: Ensure a cluster is disabled
+  cs_cluster:
     name: kvm-cluster-01
     zone: ch-zrh-ix-01
     state: disabled
+  delegate_to: localhost
 
-# Ensure a cluster is enabled
-- local_action:
-    module: cs_cluster
+- name: Ensure a cluster is enabled
+  cs_cluster:
     name: kvm-cluster-01
     zone: ch-zrh-ix-01
     state: enabled
+  delegate_to: localhost
 
-# Ensure a cluster is absent
-- local_action:
-    module: cs_cluster
+- name: Ensure a cluster is absent
+  cs_cluster:
     name: kvm-cluster-01
     zone: ch-zrh-ix-01
     state: absent
+  delegate_to: localhost
 '''
 
 RETURN = '''

--- a/lib/ansible/modules/cloud/cloudstack/cs_configuration.py
+++ b/lib/ansible/modules/cloud/cloudstack/cs_configuration.py
@@ -54,33 +54,33 @@ extends_documentation_fragment: cloudstack
 '''
 
 EXAMPLES = '''
-# Ensure global configuration
-- local_action:
-    module: cs_configuration
+- name: Ensure global configuration
+  cs_configuration:
     name: router.reboot.when.outofband.migrated
     value: false
+  delegate_to: localhost
 
-# Ensure zone configuration
-- local_action:
-    module: cs_configuration
+- name: Ensure zone configuration
+  cs_configuration:
     name: router.reboot.when.outofband.migrated
     zone: ch-gva-01
     value: true
+  delegate_to: localhost
 
-# Ensure storage configuration
-- local_action:
-    module: cs_configuration
+- name: Ensure storage configuration
+  cs_configuration:
     name: storage.overprovisioning.factor
     storage: storage01
     value: 2.0
+  delegate_to: localhost
 
-# Ensure account configuration
-- local_action:
-    module: cs_configuration
+- name: Ensure account configuration
+  cs_configuration:
     name: allow.public.user.templates
     value: false
     account: acme inc
     domain: customers
+  delegate_to: localhost
 '''
 
 RETURN = '''

--- a/lib/ansible/modules/cloud/cloudstack/cs_disk_offering.py
+++ b/lib/ansible/modules/cloud/cloudstack/cs_disk_offering.py
@@ -107,27 +107,27 @@ extends_documentation_fragment: cloudstack
 
 EXAMPLES = '''
 - name: Create a disk offering with local storage
-  local_action:
-    module: cs_disk_offering
+  cs_disk_offering:
     name: small
     display_text: Small 10GB
     disk_size: 10
     storage_type: local
+  delegate_to: localhost
 
 - name: Create or update a disk offering with shared storage
-  local_action:
-    module: cs_disk_offering
+  cs_disk_offering:
     name: small
     display_text: Small 10GB
     disk_size: 10
     storage_type: shared
     storage_tags: SAN01
+  delegate_to: localhost
 
 - name: Remove a disk offering
-  local_action:
-    module: cs_disk_offering
+  cs_disk_offering:
     name: small
     state: absent
+  delegate_to: localhost
 '''
 
 RETURN = '''

--- a/lib/ansible/modules/cloud/cloudstack/cs_domain.py
+++ b/lib/ansible/modules/cloud/cloudstack/cs_domain.py
@@ -50,22 +50,22 @@ extends_documentation_fragment: cloudstack
 
 EXAMPLES = '''
 - name: Create a domain
-  local_action:
-    module: cs_domain
+  cs_domain:
     path: ROOT/customers
     network_domain: customers.example.com
+  delegate_to: localhost
 
 - name: Create another subdomain
-  local_action:
-    module: cs_domain
+  cs_domain:
     path: ROOT/customers/xy
     network_domain: xy.customers.example.com
+  delegate_to: localhost
 
 - name: Remove a domain
-  local_action:
-    module: cs_domain
+  cs_domain:
     path: ROOT/customers/xy
     state: absent
+  delegate_to: localhost
 '''
 
 RETURN = '''

